handout/architect/service-outline.md

```diff
@@ -0,0 +1,49 @@
+# Services
+
+- Prelude (before giving you ivory tower mandates, let's look at why we might want services)
+
+- Code:
+  - Component with http calls in methods  
+    (~8 lines w/ ellipses)
+  - Point out (bullets or a sentence): Maintenance; reuse with different endpoints; caching data; complexity management
+
+- Talk about Refactoring by pulling self-contained processes into their own functions with arguments
+
+- Code:
+  - function that represents previous call, with data
+    (~3 lines)
+  - method which calls the function, passing data
+    (~5 lines w/ ellipses)
+  - Point out: testable; configurable; maintainable; still not reusable
```

Comment:
Because it's still sitting as a procedural function, outside of the class, but inside the component class-file.
Technically, you could, once you get it to this point, but you have discoverability, maintenance, idiosyncrasy concerns. The thought is that from the code standpoint, you're sufficiently done. From an organization and architecture standpoint, there are other steps required (pulling the function out of the file, and importing it as a dependency vs referencing it directly in the method code, coupled by file location, rather than as injected).